src/gui/banlistoptions.cpp

```diff
@@ -0,0 +1,104 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "banlistoptions.h"
+#include "ui_banlistoptions.h"
+
+#include <QMessageBox>
+#include <QHostAddress>
+
+#include "base/bittorrent/session.h"
+#include "base/utils/net.h"
+
+BanListOptions::BanListOptions(QWidget *parent)
+    : QDialog(parent)
+    , m_ui(new Ui::BanListOptions)
+    , m_modified(false)
+{
+    m_ui->setupUi(this);
+    m_ui->bannedIPList->addItems(BitTorrent::Session::instance()->bannedIPs());
+    m_ui->pb_BanIP->setEnabled(false);
+}
+
+BanListOptions::~BanListOptions()
+{
+    delete m_ui;
+}
+
+void BanListOptions::on_buttonBox_accepted()
+{
+    if (m_modified) {
+        // save to session
+        QStringList IPList;
+        for (int i = 0; i < m_ui->bannedIPList->count(); ++i)
+            IPList << m_ui->bannedIPList->item(i)->text();
+        BitTorrent::Session::instance()->setBannedIPs(IPList);
+    }
+    accept();
+}
+
+void BanListOptions::banIP(const QString &ip)
+{
+    if (!Utils::Net::isValidIP(ip)) {
+        QMessageBox::warning(this, tr("Warning"), tr("The entered IP address is invalid."));
+        return;
+    }
+    // the same IPv6 addresses could be written in different forms;
+    // QHostAddress::toString() result format follows RFC5952;
+    // thus we avoid duplicate entries pointing to the same address
+    QString s = QHostAddress(ip).toString();
+    QList<QListWidgetItem *> findres = m_ui->bannedIPList->findItems(s, Qt::MatchExactly);
+    if (!findres.isEmpty()) {
+        QMessageBox::warning(this, tr("Warning"), tr("The entered IP is already banned."));
+        return;
+    }
+    m_ui->bannedIPList->addItem(s);
+    m_ui->ed_IP->clear();
+    m_modified = true;
```

Comment:
It is unnecessary because it is done in `BanListOptions::on_txtIP_textChanged()` event handler called because of  `m_ui->txtIP->clear();` call.